### ISSUE TITLE
[Bug]: Changing the default `cutoff` in `MEGNet` yields inconsistent results with `.predict_structure()`

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [materialsvirtuallab/matgl#612]
### Version

12ee695

### Which OS(es) are you using?

- [ ] MacOS
- [ ] Windows
- [x] Linux

### What happened?

When training a `MEGNet` model with a `cutoff` value that is not the default value of 4.0, the predictions made during training (e.g. to produce the validation MAE) and the predictions made with `.predict_structure()` are not the same.

Take the example code below. When `cutoff=4.0`, the `val_MAE` reported by PyTorch Lightning is 0.676, and the value reported from calling `.predict_structure()` on the validation set is 0.6756802373510791, as expected. However, when `cutoff=6.0`, the `val_MAE` reported by PyTorch Lightning is 0.460, whereas the value reported from calling `.predict_structure()` on the validation set is 0.5325856619459584.

[data.json](https://github.com/user-attachments/files/21023229/data.json)

```python

import lightning as L
import torch
from dgl.data.utils import split_dataset
from lightning.pytorch.loggers import CSVLogger
from matgl.ext.pymatgen import Structure2Graph, get_element_list
from matgl.graph.data import MGLDataLoader, MGLDataset, collate_fn_graph
from matgl.layers import BondExpansion
from matgl.models import MEGNet
from matgl.utils.training import ModelLightningModule
from monty.serialization import loadfn
from pymatgen.core import Structure
from sklearn.metrics import mean_absolute_error
import warnings

warnings.simplefilter("ignore")
cutoff = 6.0

# Load dataset
data = loadfn("data.json")
structures = []
mp_ids = []
eform_per_atom = []
for entry in data:
    structures.append(Structure.from_str(entry["structure"], fmt="cif"))
    mp_ids.append(entry["material_id"])
    eform_per_atom.append(entry["formation_energy_per_atom"])

# Make model
elem_list = get_element_list(structures)
converter = Structure2Graph(element_types=elem_list, cutoff=cutoff)
mp_dataset = MGLDataset(
    structures=structures,
    labels={"Eform": eform_per_atom},
    converter=converter,
)

train_data, val_data, test_data = split_dataset(
    mp_dataset,
    frac_list=[0.8, 0.1, 0.1],
    shuffle=False,
    random_state=42,
)
train_loader, val_loader, test_loader = MGLDataLoader(
    train_data=train_data,
    val_data=val_data,
    test_data=test_data,
    collate_fn=collate_fn_graph,
    batch_size=2,
    num_workers=0,
)

node_embed = torch.nn.Embedding(len(elem_list), 16)
bond_expansion = BondExpansion(
    rbf_type="Gaussian", initial=0.0, final=5.0, num_centers=100, width=0.5
)

model = MEGNet(
    dim_node_embedding=16,
    dim_edge_embedding=100,
    dim_state_embedding=2,
    nblocks=3,
    hidden_layer_sizes_input=(64, 32),
    hidden_layer_sizes_conv=(64, 64, 32),
    nlayers_set2set=1,
    niters_set2set=2,
    hidden_layer_sizes_output=(32, 16),
    is_classification=False,
    activation_type="softplus2",
    bond_expansion=bond_expansion,
    element_types=elem_list,
    cutoff=cutoff,
    gauss_width=0.5,
)

lit_module = ModelLightningModule(model=model)

logger = CSVLogger("logs", name="MEGNet_training")
trainer = L.Trainer(max_epochs=1, logger=logger)
trainer.fit(
    model=lit_module, train_dataloaders=train_loader, val_dataloaders=val_loader
)

val_structures = [structures[i] for i in val_data.indices]
val_eform_per_atom = [eform_per_atom[i] for i in val_data.indices]

output = [model.predict_structure(structure).item() for structure in val_structures]
mae = mean_absolute_error(val_eform_per_atom, output)

print(f"calculated manual MAE {mae}")
```

**Comment by Andrew-S-Rosen** [comment]
Closing because this was a user error in that it was re-reading the cached graphs (although that is a dangerous foot-gun).
